File: project/angular_project/src/app/models/sensor-maintenance.ts
import { Sensor } from "./sensor";

export class SensorMaintenance {
  sensorMaintenance_id: number;
  installation_date: Date;
  calibration_date: Date;
  status: string;
  maintenance_by: string;
  sensor: Sensor;

  constructor() {
    this.sensorMaintenance_id = 0;
    this.installation_date = new Date();
    this.calibration_date = new Date();
    this.status = '';
    this.maintenance_by = '';
    this.sensor = new Sensor();
  }

}

File: src/main/java/com/example/project/entities/SensorMaintenance.java
package com.example.project.entities;

import java.io.Serializable;
import java.util.Date;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;


@Data
@Document(collection = "SensorMaintenance")
public class SensorMaintenance implements Serializable {
	@Id
	private int sensorMaintenance_id;
	private Date installation_date;
	private Date calibration_date;
	private String status;
	private String maintenance_by;
    private Sensor sensor;
	
}